src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts

```diff
@@ -28,17 +32,62 @@ const query = gql`
   }
 `;
 
+export type YieldYakFarmDetails = {
+  address: string;
+  deployed: number;
+  name: string;
+  depositToken: {
+    address: string;
+    symbol: string;
+    decimals: number;
+  };
+  totalDeposits: string;
+};
+
 @Injectable()
 export class YieldYakVaultTokenDefinitionsResolver {
-  constructor(@Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit) {}
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(YieldYakContractFactory) private readonly yieldyakContractFactory: YieldYakContractFactory,
+  ) {}
 
   @Cache({
     key: network => `studio:yield-yak:${network}:vault-data`,
     ttl: 5 * 60, // 60 minutes
   })
-  async getVaultDefinitionsData() {
-    const endpoint = 'https://api.thegraph.com/subgraphs/name/yieldyak/reinvest-tracker';
-    const data = await this.appToolkit.helpers.theGraphHelper.request<FarmsResponse>({ endpoint, query });
-    return data.farms;
+  async getVaultDefinitionsData(network: Network): Promise<FarmsResponse['farms']> {
+    const farms = await Axios.get<YieldYakFarmDetails[]>('https://staging-api.yieldyak.com/farms').then(x => x.data);
+    const multicall = this.appToolkit.getMulticall(network);
+
+    const response = await Promise.all(
+      farms.map(async farm => {
+        const farmContract = this.yieldyakContractFactory.yieldYakVault({
+          address: farm.address,
+          network: Network.AVALANCHE_MAINNET,
```

Comment:
can reuse `network` from the fn params here

Reply:
nevermind you can remove the contract altogether as referred in the other comment!